staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/README.md

```diff
@@ -0,0 +1,68 @@
+# Azure Active Directory plugin for client authentication
+
+This plugin provides an integration with [Azure CLI](https://github.com/Azure/azure-cli) and Azure Active Directory device flow.
+
+## Usage
+
+### Microsoft Azure CLI 2.0
+
+1. Configure the `apiserver` to use the Azure Active Directory as an OIDC provider with following options
+
+   ```
+   --oidc-client-id="https://management.core.windows.net/" \
+   --oidc-issuer-url="https://sts.windows.net/TENANT_ID/"
+   --oidc-username-claim="sub"
+   ```
+
+   * Replace `TENANT_ID` with your tenant ID.
+
+2. Login with Azure CLI
+
+  ```
+  az login
+  ```
+
+3. Configure the `kubectl` to use the `azure` authentication provider
+
+   ```
+   kubectl config set-credentials "USER_NAME" --auth-provider=azure
+   ```
+
+   * Replace `USER_NAME` with your user name.
+
+### Device Flow
```

Comment:
Unfortunately I couldn't find some reasonable documentation for this flow. The big difference between client credentials and device code flows is that the JWT token acquired with the former one contains user information. It is a user token not a client token.  The claims form this token can be used later for role-based access control.

This flow is suited for CLI tools. e.g. Azure CLI makes use of it in order to acquire user tokens.